### PR TITLE
 Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by torobucci**
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/like_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  subject { Like.new(user_id: 1, post_id: 1) }
+
+  before { subject.save }
+
+  it 'new like should be saved in the database' do
```
Comment (Davidosky007) [inline spec/like_spec.rb:8]: ## * Kindly Add unit specs for all of your models' methods as unit tests for them are missing ( for example you have the method `update_post_likes_counter` which needs to be tested ) 
image

File: spec/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  it 'title should be present' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'title should not exceed 250 characters' do
+    subject.title = 'a' * 255
+    expect(subject).to_not be_valid
+  end
```
Comment (Davidosky007) [inline spec/post_spec.rb]: ## * Kindly Add unit specs for all of your models' methods as unit tests for them are missing ( for example you have the method `recent_comments` ` update_user_posts_counter` which needs to be tested ) 
image

File: spec/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  subject { Comment.new(text: 'Nice post', user_id: 1, post_id: 1) }
+
+  before { subject.save }
+
+  it 'new comment should be saved in the database' do
+    expect(subject.new_record?).to be_truthy
```
Comment (Davidosky007) [inline spec/comment_spec.rb:9]: ## * Kindly Add unit specs for all of your models' methods as unit tests for them are missing ( for example you have the method `update_post_comments_counter` which needs to be tested ) 
image

File: spec/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Tom', posts_counter: 3) }
+  before { subject.save }
+
+  it 'name should be present' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+  it 'posts_counter should be an integer greater than or equal to zero' do
```
Comment (Davidosky007) [inline spec/user_spec.rb]: ## * Kindly Add unit specs for all of your models' methods as unit tests for them are missing ( for example you have the method `recent_posts` ` update_user_posts_counter` which needs to be tested ) 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @torobucci 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* All tests passed 👍 
* Validations for models used 👍 
* Unit tests for models written 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @torobucci 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done adding the requested unit tests for the custom methods

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**